mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +230,140 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
```

Comment:
Btw, now that there is no public setter, let's make this a val and pass it to a private constructor.